deeprankcore/dataset.py

```diff
@@ -50,132 +50,99 @@ def save_hdf5_keys(
+        edge_features: Union[List[str], str] = "all",
+        clustering_method: str = "mcl",
         classes: List = None,
         tqdm: bool = True,
-        subset: list = None,
-        node_feature: Union[List[str], str] = "all",
-        edge_feature: Union[List[str], str] = "all",
-        clustering_method: str = "mcl",
-        edge_feature_transform: Callable = lambda x: np.tanh(-x / 2 + 2) + 1,
+        root: str = "./",
+        transform: Callable = None,
+        pre_transform: Callable = None,
+        edge_features_transform: Callable = lambda x: np.tanh(-x / 2 + 2) + 1,
+        target_filter: dict = None,
     ):
         """Class from which the hdf5 datasets are loaded.
 
         Args:
-            root (str, optional): Root directory where the dataset should be
-            saved. Defaults to "./"
-
-            hdf5_path (str, optional): Path to hdf5 file(s). For multiple hdf5 files, 
-            insert the paths in a list. Defaults to None.
+            hdf5_path (str or list): Path to hdf5 file(s). For multiple hdf5 files, 
+                insert the paths in a list. Defaults to None.
 
-            transform (callable, optional): A function/transform that takes in
-            a torch_geometric.data.Data object and returns a transformed version.
-            The data object will be transformed before every access. Defaults to None.
-
-            pre_transform (callable, optional):  A function/transform that takes in
-            a torch_geometric.data.Data object and returns a transformed version.
-            The data object will be transformed before being saved to disk. Defaults to None.
-
-            dict_filter (dictionary, optional): Dictionary of type [name: cond] to filter the molecules.
-            Defaults to None.
+            subset (list, optional): list of keys from hdf5 file to include. Defaults to None (meaning include all).
 
-            target (str, optional): irmsd, lrmsd, fnat, bin, capri_class or dockq. It can also be a custom-defined
-            target given to the Query class as input (see: deeprankcore.query); in the latter case, specify
-            here its name. Only numerical target variables are supported, not categorical. If the latter is your case,
-            please convert the categorical classes into numerical class indices before defining the HDF5DataSet instance.
-            Defaults to None.
+            target (str, optional): default options: irmsd, lrmsd, fnat, bin, capri_class or dockq. 
+                It can also be a custom-defined target given to the Query class as input (see: deeprankcore.query); 
+                in this case, the task parameter needs to be explicitly specified as well.
+                Only numerical target variables are supported, not categorical. If the latter is your case, please convert 
+                the categorical classes into numerical class indices before defining the GraphDataset instance.
+                Defaults to None.
```

Comment:
from issue #152:
> After having decided about the above, and implemented that, remember to add an example code in the readme of target parameter usage.

Is this truly necessary? I think the docstring as is should be sufficient:

Reply:
No it's not necessary, and in case I'll add it with #227